File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l, r = 1, max(piles)

        while l <= r:
            mid = (l+r) // 2
            totalTimes = 0

            for p in piles:
                totalTimes += math.ceil(float(p) / mid)
            if totalTimes <= h:
                res = mid
                r = mid - 1
            else:
                l = mid + 1

        return res 

File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-2.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int l= 0, res = 0;

        for (int r=0; r < nums.length; r++){
            if (nums[r] == 0) {
                k -= 1;
            }

            while (k < 0){
                k += (nums[l] == 0 ? 1 : 0);
                l++;
            }

            res = Math.max(res, r - l + 1);
        }

        return res;
    }
}